internal/api/hooks.go

```diff
@@ -152,14 +152,8 @@ func (a *API) runHTTPHook(r *http.Request, hookConfig conf.ExtensibilityPointCon
 			if rsp.Body == nil {
 				return nil, nil
 			}
-			contentLength := rsp.ContentLength
-			if contentLength == -1 {
-				return nil, unprocessableEntityError(ErrorCodeHookPayloadUnknownSize, "Payload size not known")
-			}
-			if contentLength >= PayloadLimit {
-				return nil, unprocessableEntityError(ErrorCodeHookPayloadOverSizeLimit, fmt.Sprintf("Payload size is: %d bytes exceeded size limit of %d bytes", contentLength, PayloadLimit))
-			}
```

Comment:
We should still keep this, as when we do know what the content-length is we don't have to waste resources reading a body.

Reply:
hmm ok i think a more reliable solution would be to attempt to read the response body again to check for excess and return this error

Reply:
Can consider checking this:

[`a156a5b` (#1474)](https://github.com/supabase/auth/pull/1474/commits/a156a5b6d4848cd45cbd442a737cda19363e42ca#diff-c3767c4f85e28d21d8d8a14e8cf34c49d88ee88362444ba2217856cddb9e66aeR77)

Reply:
ah yeah that's what i'm gonna do